src/search/parser/lexical_analyzer.h

```diff
@@ -0,0 +1,42 @@
+#ifndef PARSER_LEXICAL_ANALYZER_H
+#define PARSER_LEXICAL_ANALYZER_H
+
+#include <ostream>
+#include <string>
+#include <vector>
+
+namespace parser {
+enum class TokenType {
+    OPENING_PARENTHESIS,
+    CLOSING_PARENTHESIS,
+    OPENING_BRACKET,
+    CLOSING_BRACKET,
+    COMMA,
+    EQUALS,
+    INTEGER,
+    FLOAT,
+    BOOLEAN,
+    IDENTIFIER,
+    LET
+};
+
+struct Token {
+    std::string content;
+    TokenType type;
+
+    friend std::ostream &operator<<(std::ostream &out, const Token &token);
```

Comment:
Without `friend`, this produces an error that the function must take exactly one argument. Should this be defined outside of the struct then? I like having this in the struct because when looking at the code, I can see more clearly that I only have to care about this when I care about the class `Token`.